src/main/services/ProjectService.ts

```diff
@@ -1,34 +1,55 @@
 import log from 'electron-log';
 import {
   ExtractFromProjectDTO,
-  INewProject, Inventory,
+  INewProject,
+  Inventory,
   InventoryKnowledgeExtraction,
   ProjectState,
-  ReuseIdentificationTaskDTO
+  ReuseIdentificationTaskDTO,
 } from '../../api/types';
-import {Project} from '../workspace/Project';
-import {workspace} from '../workspace/Workspace';
-import {modelProvider} from './ModelProvider';
-import {Scanner} from '../task/scanner/types';
-import {userSettingService} from './UserSettingService';
-import {ProjectFilterPath} from '../workspace/filters/ProjectFilterPath';
-import {CodeScannerPipelineTask} from '../task/scanner/scannerPipeline/CodeScannerPipelineTask';
-import {ScannerPipelineFactory} from "../task/scanner/scannerPipelineFactory/ScannerPipelineFactory";
-import ScannerType = Scanner.ScannerType;
-import ScannerMode = Scanner.ScannerMode;
-import {ProjectKnowledgeExtractor} from "../modules/projectKnowledge/ProjectKnowledgeExtractor";
+import { Project } from '../workspace/Project';
+import { workspace } from '../workspace/Workspace';
+import { modelProvider } from './ModelProvider';
+import { Scanner } from '../task/scanner/types';
+import { userSettingService } from './UserSettingService';
+import { ProjectFilterPath } from '../workspace/filters/ProjectFilterPath';
+import { CodeScannerPipelineTask } from '../task/scanner/scannerPipeline/CodeScannerPipelineTask';
+import { ScannerPipelineFactory } from '../task/scanner/scannerPipelineFactory/ScannerPipelineFactory';
+import { ProjectKnowledgeExtractor } from '../modules/projectKnowledge/ProjectKnowledgeExtractor';
 import { ReuseIdentificationTask } from '../task/reuseIdentification/ReuseIdentificationTask';
+import ScannerMode = Scanner.ScannerMode;
+import ScannerType = Scanner.ScannerType;
 
 class ProjectService {
+  private setCryptographyScanType(p: Project): void {
+    if (p.getGlobalApiKey()) {
+      const types = p.metadata.getScannerConfig().type;
+      types.push(ScannerType.CRYPTOGRAPHY);
+      const uniqueTypes = new Set(types);
+      p.metadata.getScannerConfig().type = Array.from(uniqueTypes);
+    } else {
+      p.metadata.getScannerConfig().type.filter((t) => t !== ScannerType.CRYPTOGRAPHY);
```

Comment:
In case the user has setted the API Key we add the Crypto scan . We avoid duplicated types with the Set.
Otherwise , when user hasn't the API Token we need to be sure the Cypto scan be removed.